samples/ProjectGrandmaster.Unity/Assets/Script/Board/BoardInformation.cs

```diff
@@ -0,0 +1,633 @@
+        public bool PieceHandling { get; set; }
+
+        // GameObject related to displaying end result
+        public GameObject EndGameResult;
+        public GameObject whiteText;
+        public GameObject blackText;
+        public GameObject leftText;
+        public GameObject rightText;
+
+        // Piece cannot be moved until previous move is complete
+        public bool CanMove { get; set; }
+
+        public bool Check { get; set; }
+        public bool GameEnded { get; set; }
+        public bool AIEnabled { get; set; }
+        public bool ghostActive { get; set; }
+
+        // Side that won. -1 if black, 0 if draw, +1 if white.
+        // Game Ended must also be true
+        public int Winner { get; set; }
+
+        // Arrow on the boardmenu that let's the player know if its their turn
+        // Green if true, else red. 
+        public GameObject whiteArrow;
+        public GameObject blackArrow;
+        public Material playerTurn;
+        public Material opponentTurn;
+
+        // Variables related to pawn promotion
+        public Mesh mesh { get; set; }
+        public bool promoted { get; set; }
+        public bool meshChosen { get; set; }
+        public GameObject pawnPromo;
+
+        public GameObject boardMenuTileText;
+        public GameObject boardMenuTileTextTwo;
+        public Mesh pawnMesh;
+
+        void Start()
+        {
+            /// <summary>
+            /// 2D chess layout with individual piece GameObject
+            /// </summary>
+            board = new GameObject[8, 8];
+            piecesOnBoard = new List<GameObject>();
+
+            foreach (GameObject piece in GameObject.FindGameObjectsWithTag("pieces"))
+            {
+                PieceInformation info = piece.GetComponent<PieceInformation>();
+                int x = info.GetXPosition();
+                int z = info.GetZPosition();
+                board[z, x] = piece;
+
+                /// <summary>
+                /// Initialise kings
+                /// </summary>
+                if (info.type == PieceInformation.Type.King)
+                {
+                    if (info.colour == PieceInformation.Colour.White) { whiteKing = piece; }
+                    else { blackKing = piece; }
+                }
+
+                /// <summary>
+                /// Add piece to piecesOnBoard list to keep track of all pieces on board
+                /// </summary>
+                piecesOnBoard.Add(piece);
+            }
+
+            /// <summary>
+            /// Assign null for empty positions
+            /// </summary>
+            for (int i = 2; i <= 5; i++)
+            {
+                for (int j = 0; j <= 7; j++)
+                {
+                    board[i, j] = null;
+                }
+            }
+            /// <summary>
+            /// Initialise whose turn it is
+            /// </summary>
+            turn = Colours.White;
+
+            /// <summary>
+            /// Initialise Data Structure
```

Comment:
So, this is interesting: Initialise is Australian, and Initialize is US. I am fine with leaving either one.

But, could you lower case `data structure`